src/python_testing/TC_EWATERHTRM_2_1.py

```diff
@@ -0,0 +1,163 @@
+            TestStep(7, "Send ChangeToMode command with NewMode"),
+            TestStep(8, "Read CurrentMode attribute"),
+            TestStep(9, "Manually put the device in a state from which it will SUCCESSFULLY transition"),
+            TestStep(10, "Read CurrentMode attribute"),
+            TestStep(11, "Send ChangeToMode command with NewMode"),
+            TestStep(12, "Read CurrentMode attribute"),
+            TestStep(13, "Send ChangeToMode command with NewMode set to an invalid mode"),
+            TestStep(14, "Read CurrentMode attribute"),
+        ]
+        return steps
+
+    async def read_mode_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.WaterHeaterMode
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    async def send_change_to_mode_cmd(self, newMode) -> Clusters.Objects.WaterHeaterMode.Commands.ChangeToModeResponse:
+        ret = await self.send_single_cmd(cmd=Clusters.Objects.WaterHeaterMode.Commands.ChangeToMode(newMode=newMode), endpoint=self.endpoint)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.WaterHeaterMode.Commands.ChangeToModeResponse),
+                            "Unexpected return type for Water Heater Mode ChangeToMode")
+        return ret
+
+    def pics_TC_EWATERHTRM_2_1(self) -> list[str]:
+        return ["EWATERHTRM.S"]
+
+    @async_test_body
+    async def test_TC_EWATERHTRM_2_1(self):
+
+        # Valid modes. Only ModeManual referred to in this test
+        # ModeOff    = 0
+        ModeManual = 1
+        # ModeTimed  = 2
+
+        self.endpoint = self.matter_test_config.endpoint
+
+        attributes = Clusters.WaterHeaterMode.Attributes
+
+        self.step(1)
+        # Commission DUT - already done
+
+        self.step(2)
+
+        supported_modes = await self.read_mode_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.SupportedModes)
+
+        logging.info(f"SupportedModes: {supported_modes}")
+
+        asserts.assert_greater_equal(len(supported_modes), 2,
+                                     "SupportedModes must have at least two entries!")
+
+        self.step(3)
+
+        old_current_mode = await self.read_mode_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.CurrentMode)
+
+        logging.info(f"CurrentMode: {old_current_mode}")
+
+        # pick a value that's not on the list of supported modes
+        modes = [m.mode for m in supported_modes]
+        invalid_mode = max(modes) + 1
+
+        self.step(4)
+
+        ret = await self.send_change_to_mode_cmd(newMode=old_current_mode)
+        logging.info(f"ret.status {ret.status}")
+        asserts.assert_equal(ret.status, Status.Success,
+                            "Changing the mode to the current mode should be a no-op")
+
+        # Steps 5-9 are not performed as EWATERHTRM.S.M.CAN_TEST_MODE_FAILURE is false
```

Comment:
Is that always false?

Reply:
I think this is a copy and paste from another test case - but there are only 8 steps in the WHM_2_1 test plan so the TestSteps above don't match the test plan.

Reply:
We do not support this currently. It's not clear to me we will support it but propose to follow-up in https://github.com/project-chip/connectedhomeip/issues/34565 and leave as is for the time being.